releases/XCFramework/libzcashlc.xcframework/ios-arm64/libzcashlc.framework/Headers/zcashlc.h

```diff
@@ -52,6 +52,89 @@ typedef struct FFIEncodedKeys {
   uintptr_t len;
 } FFIEncodedKeys;
 
+/**
+ * A struct that contains a subtree root.
+ *
+ * # Safety
+ *
+ * - `root_hash_ptr` must be non-null and must be valid for reads for `root_hash_ptr_len`
+ *   bytes, and it must have an alignment of `1`.
+ * - The total size `root_hash_ptr_len` of the slice pointed to by `root_hash_ptr` must
+ *   be no larger than `isize::MAX`. See the safety documentation of `pointer::offset`.
+ */
+typedef struct FfiSubtreeRoot {
+  uint8_t *root_hash_ptr;
+  uintptr_t root_hash_ptr_len;
+  uint32_t completing_block_height;
+} FfiSubtreeRoot;
+
+/**
+ * A struct that contains a pointer to, and length information for, a heap-allocated
+ * slice of [`FfiSubtreeRoot`] values.
+ *
+ * # Safety
+ *
+ * - `ptr` must be non-null and must be valid for reads for `len * mem::size_of::<FfiSubtreeRoot>()`
+ *   many bytes, and it must be properly aligned. This means in particular:
+ *   - The entire memory range pointed to by `ptr` must be contained within a single
+ *     allocated object. Slices can never span across multiple allocated objects.
+ *   - `ptr` must be non-null and aligned even for zero-length slices.
+ *   - `ptr` must point to `len` consecutive properly initialized values of type
+ *     [`FfiSubtreeRoot`].
+ * - The total size `len * mem::size_of::<FfiSubtreeRoot>()` of the slice pointed to
+ *   by `ptr` must be no larger than isize::MAX. See the safety documentation of
+ *   `pointer::offset`.
+ * - See the safety documentation of [`FfiSubtreeRoot`]
+ */
+typedef struct FfiSubtreeRoots {
+  struct FfiSubtreeRoot *ptr;
+  uintptr_t len;
+} FfiSubtreeRoots;
+
+/**
+ * A struct that contains details about scan progress.
+ *
+ * When `denominator` is zero, the numerator encodes a non-progress indicator:
+ * - 0: progress is unknown.
+ * - 1: an error occurred.
```

Comment:
Another way to do this would be to call the struct FfiScanProgressResult, and have another field that contains a status enum (None, Some, or Error).